src/main/habit-modal-dialogue/habit-modal-dialogue.component.ts

```diff
@@ -0,0 +1,211 @@
+    const today = new Date();
+    const firstDateOfMonth = new Date(today.getFullYear(), today.getMonth(), 1);
+    this.selectedDates.push(firstDateOfMonth);
+  }
+
+  close() {
+    this.ref.close();
+  }
+
+  getIconSource(): string {
+    return iconMap[this.habitName] || 'assets/svg/mark.svg';
+  }
+
+  checkDays(day: string) {
+    return this.days.includes(day);
+  }
+
+  updateIntervalPerDays(selectedInterval: string) {
+    this.intervalPerDays = selectedInterval;
+    this.checkRepeat = AppConstants.interval;
+  }
+
+  toggleDay(day: string): void {
+    const index = this.days.indexOf(day);
+
+    if (index !== -1 && this.days.length > 1) {
+      // If the day is present and there is more than one day, remove it
+      this.days.splice(index, 1);
+    } else if (index === -1) {
+      // If the day is not present, add it
+      this.days.push(day);
+    }
+    this.checkRepeat = AppConstants.days;
+  }
+
+  updateGoalFrequency(selectedFrequency: string) {
+    this.goalFrequency = selectedFrequency;
+  }
+
+  getDayValue(): string {
+    const allDays = Object.values(daysOfWeek);
+    if (this.checkRepeat == AppConstants.days) {
+      if (this.areArraysEqual(this.days, allDays)) {
+        this.repeatValue = AppConstants.Daily;
+        return AppConstants.Daily;
+      } else {
+        this.repeatValue = this.days.join(' ');
+        const selectedDays = this.days.map(day => day.substring(0, 3)); // Abbreviate to three letters
+        return selectedDays.join(',');
+      }
+    } else if (this.checkRepeat == AppConstants.interval) {
+      this.repeatValue = this.intervalPerDays;
+      return this.intervalPerDays;
+    } else {
+      this.repeatValue = this.selectedDates.map(date => date.toLocaleDateString('en-US')).join(',');
+      return this.months;
+    }
+  }
+
+  areArraysEqual(firstArray: string[], secondArray: string[]): boolean {
+    if (firstArray.length !== secondArray.length) {
+      return false;
+    }
+
+    const sortedArr1 = firstArray.slice().sort();
+    const sortedArr2 = secondArray.slice().sort();
+
+    return sortedArr1.every((value, index) => value === sortedArr2[index]);
+  }
+
+  updateFrequencyPeriod(selectedFrequency: string) {
+    this.frequencyPerPeriod = selectedFrequency;
+  }
+
+  updateTimeOfDay(selectedTime: TimeOfDay) {
+    const index = this.timeOfDay.indexOf(selectedTime);
+
+    if (index !== -1) {
+      // If the time is already in the array, check if there are more than one item
+      if (this.timeOfDay.length > 1) {
+        // If there is more than one item, remove it
+        this.timeOfDay.splice(index, 1);
+      }
+      // If there is only one item, do nothing (prevent removal)
+    } else {
+      // If the time is not in the array, add it
+      this.timeOfDay.push(selectedTime);
+    }
+  }
+
+  isTimeOfDaySelected(time: TimeOfDay): boolean {
+    return this.timeOfDay.includes(time);
+  }
+
+  getTimeOfDayDisplayValue(timeOfDay: TimeOfDay[]): string {
```

Comment:
please add comments to explain the purpose of a function which has complex logic

Reply:
updated